conference.js

```diff
@@ -1637,9 +1637,27 @@ export default {
 
         if (!this.localPresenterVideo && !mute) {
             // create a new presenter track and apply the presenter effect.
-            const { height } = this.localVideo.track.getSettings();
-            const defaultCamera
-                = getUserSelectedCameraDeviceId(APP.store.getState());
+            let { height } = this.localVideo.track.getSettings();
+
+            // Workaround for Firefox since it doesn't return the correct width/height of the desktop stream
+            // that is being currently shared.
+            if (!height || height === 0) {
+                const desktopResizeConstraints = {
+                    width: 1280,
+                    height: 720,
+                    resizeMode: 'crop-and-scale'
+                };
+
+                try {
+                    await this.localVideo.track.applyConstraints(desktopResizeConstraints);
```

Comment:
is `applyConstraints` supported everywhere we need it to be?

Reply:
`applyConstraints` is supported from Chrome 63, Firefox 50 onwards and on Safari.